include/deal.II/numerics/data_out_resample.h

```diff
@@ -0,0 +1,115 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2021 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_data_out_resample_h
+#define dealii_data_out_resample_h
+
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/data_out_dof_data.h>
+
+#include <memory>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * A DataOut class that does not output the result on the original triangulation
+ * but interpolates the result onto a second triangulation. By using this class,
+ * one output the result obtained on an unstructured mesh onto a structures one
+ * or one can create a slice in 3D.
+ */
+template <int dim, int patch_dim, int spacedim>
+class DataOutResample
+  : public DataOut_DoFData<dim, patch_dim, spacedim, spacedim>
+{
+public:
+  /**
+   * Constructor taking the triangulation and mapping
+   */
+  DataOutResample(const Triangulation<patch_dim, spacedim> &patch_tria,
+                  const Mapping<patch_dim, spacedim> &      patch_mapping);
+
+  /**
+   * Update the mapping of original triangulation. One needs to call this
+   * function if the mapping has changed.
+   */
+  void
+  update_mapping(const Mapping<dim, spacedim> &mapping,
+                 const unsigned int            n_subdivisions = 0);
+
+  /**
+   * This is the central function of this class since it builds the list of
+   * patches to be written by the low-level functions of the base class. A
+   * patch is, in essence, some intermediate representation of the data on
+   * each cell of a triangulation and DoFHandler object that can then be used
+   * to write files in some format that is readable by visualization programs.
+   */
+  void
+  build_patches();
```

Comment:
I think it might be nice if this function took the same arguments as `DataOut::build_patches()`, and then just forward these arguments.